lib/modules/side_menu/side_menu_interactor.dart

```diff
@@ -1,10 +1,27 @@
 part of 'side_menu_module.dart';
 
-abstract class SideMenuInteractorDelegate {}
+abstract class SideMenuInteractorDelegate {
+  BehaviorSubject<void> get logoutDidSuccess;
+}
 
 abstract class SideMenuInteractor
-    extends Interactor<SideMenuInteractorDelegate> {}
+    extends Interactor<SideMenuInteractorDelegate> {
+  UserInfo get authenticatedUser;
+  void logout();
+}
 
 class SideMenuInteractorImpl extends SideMenuInteractor {
-  //
+  final AuthRepository _authRepository = locator.get();
+
+  @override
+  UserInfo get authenticatedUser => _authRepository.user!;
+
+  @override
+  void logout() {
+    _authRepository
+        .logout()
+        .then((value) => delegate?.logoutDidSuccess.add(null))
+        .onError<Exception>(
+            (error, stackTrace) => delegate?.logoutDidSuccess.add(null));
```

Comment:
Shouldn't we handle error instead of forcing user logout regardless of the result?

Reply:
@manh-t I think about if our system has an unexpected problem and it always throws the error, so that the user will unable to logout then they may need to uninstall the app to logout 😓

Reply:
@markgravity so that our system should be fixed. I think you should not be worry about that case 😅  Let's show a toast or sth like that when error happen 🚀

Reply:
wait @markgravity , I didn't see any changes here. Can you please check again? 🙏

Reply:
@manh-t let me commit it, still waiting for the change above. A sec 😁

Reply:
@manh-t I updated!